app/services/signature-service.js

```diff
@@ -49,14 +49,13 @@ export default class SignatureService extends Service {
       signFlow.status = status;
       await signFlow.save();
     }
-
     // Prepare sign flow: create preparation activity and send to SH
     const response = await uploadPiecesToSigninghub(signFlows);
     if (!response.ok) {
       for (let signFlow of signFlows) {
         await this.removeSignFlow(signFlow, true);
       }
-      throw new Error('Failed to upload piece to Signing Hub');
+      throw new Error(response.statusText);
```

Comment:
`response.statusText` will contain [HTTP status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) messages (e.g. `OK` for status code 200, `Internal Server Error` for 500).

In general our services (should) return a JSON body that contains a JSON:API compliant body (see the [`error` helper in the Python template](https://github.com/mu-semtech/mu-python-template/blob/master/helpers.py#L62-L77) or the [`error` handler in the Javascript template](https://github.com/mu-semtech/mu-javascript-template/blob/master/helpers/mu/server.js#L32-L37)).

I think here you should try to get the JSON body and throw that (displaying the whole JSON body is fine, no need to extract the message/detail at this time). But still bewaring that it may not be JSON (either because the service is misconfigured or because the faulty response doesn't come from the service because e.g. it's down), in which case using `response.statusText` is probably okay (though you could check if it's possible to get the response body and just show that, whatever it may be).